org.ekstep.sunbirdcommonheader-1.7/editor/headerApp.js

```diff
@@ -115,7 +116,42 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
         $scope.$safeApply();
     };
 
-    $scope.setEditorDetails = function () {
+    /*
+    * This method is used for download Table of contents which is created by Textbook creator.
+    */
+    $scope.downloadToc = function() {
+        $scope.loader = true;
+        org.ekstep.services.textbookService.downloadFile(ecEditor.getContext('contentId'), function(err, resp) {
+            if (!err && resp.data.responseCode == "OK") {
+                $scope.loader = false;
+                $scope.$safeApply();
+                ecEditor.dispatchEvent("org.ekstep.toaster:success", {
+                    title: 'Table of Content downloadeding!',
+                    position: 'topCenter',
+                    icon: 'fa fa-download'
+                });
+                var link = document.createElement('a');
```

Comment:
This should be hidden element in the template itself. You will be changing the href of the element & triggering click event from here.